### PR TITLE
The initial Web UX portion of the Upgrade for Time-Lapse Data Handling

### PR DESCRIPTION
**Pull request by sandonian**
Please see b/234031896 for a video attachment showing the current progress.

While this format is subject to change, the UX can now handle time-lapse data via new zip files in the format:

X.Zip
    config.json
    view-hierarchy-folder1/
        hierarchy.txt
        config.json
        img/
    view-hierarchy-folder2/
        hierarchy.txt
        config.json
        img/
    view-hierarchy-folder3/
        hierarchy.txt
        config.json
        img/
    ...
    ...
[Archive.zip](https://github.com/google/web-hv/files/8931222/Archive.zip) is an example zip file.

This change also includes a commit changing "var" variables to using the "let" keyword instead.

### CODE REVIEW

File: js/hview.js [hunk 1]
```diff
@@ -620,6 +620,83 @@ $(function () {
 
     }
 
+    renderMultipleHViewsAction = function (jsZip) {
+        let vListJQueries /* jQuery[] */  = []
+        let boxJQueries /* jQuery[] */ = []
+        let imageUrls /* URL[] */ = []
```
Comment (sunnygoyal) [inline js/hview.js:626]: Probably remove the imageUrl loading for now as it adds more memory overhead and we are not going to have this anyway.

File: js/hview.js [hunk 2]
```diff
@@ -620,6 +620,83 @@ $(function () {
 
     }
 
+    renderMultipleHViewsAction = function (jsZip) {
```
Comment (sunnygoyal) [inline js/hview.js:623]: this action is supposed to be reversible (user can press back and load a different file). so any UI change or newly attached event handlers should be cleared accordingly

File: js/hview.js [hunk 3]
```diff
@@ -620,6 +620,83 @@ $(function () {
...
+                })
+            }
+        }
+    
+        /* Because the input slider is only shown for multi-moment view hierarchy data,
+           there are some UX updates necessary to show the correct UX. */
+        $(".slider-group").show()
+        $("#vlist").css("bottom", "30px");
+        $("#tl-range-container").css("height", "20px")
+        $("#left-panel-divider").css("height", "10px")
```
Comment (sunnygoyal) [inline js/hview.js:676]: instead of updating sizes in code, put it behind a css class. Something like:

`root.addClass(view-slider-shown)` which overrides these sizes, so that the class can be removed easily as well

File: js/hview.js [hunk 4]
```diff
@@ -620,6 +620,83 @@ $(function () {
...
+    
+        let previousIndex = 0;
+        $(".tl-range").on("input change", (jQueryEvent) => {
```
Comment (sunnygoyal) [inline js/hview.js:679]: on multiple invocations of `renderMultipleHViewsAction`, it will add multiple handlers

File: js/ddmlib/ddmclient.js [hunk 5]
```diff
@@ -282,7 +282,7 @@ function createViewController(appInfo) {
 }
 
 function parseViewData(data, cmd, callback) {
-    let w = createWorker("js/ddmlib/worker.js");
```
Comment (sunnygoyal) [inline js/ddmlib/ddmclient.js]: why is this change here?

**Review by sunnygoyal — COMMENTED** [review]
Could you share some screenshots for desktop and mobile UI

File: js/ddmlib/ddmclient.js (hunk 5 above)
Comment (sandonian) [reply js/ddmlib/ddmclient.js]: That's a fair question. Without making this change, the multi-zip screen wouldn't load. There is an ActiveState variable that would maintain a reference to these workers (via the createWorker method) and preemptively terminate them.

File: js/hview.js (hunk 2 above)
Comment (sandonian) [reply js/hview.js:623]: I just tested this locally, and right now even regular zip files don't have a back button. That being said, this functionality should be pretty simple to add. Would you like me to in this CL?

File: js/hview.js (hunk 1 above)
Comment (sandonian) [reply js/hview.js:626]: I will remove that code from this project.

File: js/hview.js (hunk 3 above)
Comment (sandonian) [reply js/hview.js:676]: Good idea Sunny! I will do that.

File: js/hview.js (hunk 4 above)
Comment (sandonian) [reply js/hview.js:679]: In a previous comment to this one, I wrote:

"I just tested this locally, and right now even regular zip files don't have a back button. That being said, this functionality should be pretty simple to add. Would you like me to in this CL?"

On a similar line of thinking, do you think I should add support for "back" functionality in this CL?

**Comment by sandonian** [comment]
> Could you share some screenshots for desktop and mobile UI

I tried to add videos to the CL, but they were too many MB. If you look at this bug: b/234031896, there is a nice video that shows the Web UX. 

I can collect screenshots and attach them to this CL. I also haven't tested out the mobile UI yet either, but will do that too.

File: js/hview.js (hunk 2 above)
Comment (sunnygoyal) [reply js/hview.js:623]: Try opening a bug report, it should have the back button

File: js/ddmlib/ddmclient.js (hunk 5 above)
Comment (sunnygoyal) [reply js/ddmlib/ddmclient.js]: This change would cause workers to leak. You might need to figure out why the ActiveState is getting cleared and load everything before that

File: css/app.css [hunk 6]
```diff
@@ -27,8 +27,9 @@
     --title-color: #2196F3;
     --progress-color: #1976D2;
 
-    --selected-color: #3b99fc;
-    --selected-color2: rgba(59, 153, 252, 0.1);
+    --selected-color: orangered;
```
Comment (sunnygoyal) [inline css/app.css:30]: probably change the colors to go with the current color scheme. These colors look too flashy in the video
